webdriver/tests/bidi/external/speculation/conftest.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+from typing import Any, Mapping
+
+from webdriver.bidi.modules.script import ContextTarget
+
+
+@pytest.fixture
+def speculation_rules_helper(bidi_session):
+    """Helper for adding speculation rules to a page."""
+    
+    async def add_speculation_rules(context: Mapping[str, Any], rules: str):
+        """Add speculation rules script to the page."""
+        await bidi_session.script.evaluate(
+            expression=f"""
+                const script = document.createElement('script');
+                script.type = 'speculationrules';
+                script.textContent = `{rules}`;
+                document.head.appendChild(script);
+            """,
+            target=ContextTarget(context["context"]),
+            await_promise=False
+        )
+    
+    return add_speculation_rules
+
+
+@pytest.fixture
+def add_prefetch_link(bidi_session):

```

Comment:
Consider inlining this into `speculation_rules_helper` method, as they are always called together.

Reply:
After inlining this fixture is not needed anymore